test/gtest/ucp/test_ucp_proto.cc

```diff
@@ -563,4 +609,72 @@ UCS_TEST_P(test_proto_perf_add_func, intersect_middle)
     expect_empty_range(3000, SIZE_MAX);
 }
 
-UCP_INSTANTIATE_TEST_CASE_TLS(test_proto_perf_add_func, all, "all")
+UCS_TEST_P(test_proto_perf, agg2)
+{
+    /*
+     * 500   1000                  1999   2999
+     *  |     |                      |     |
+     *  +-----+------ local_tl ------+-----+
+     *        |                      |
+     *        |                      |
+     *        +------ local_cpu -----+
+     */
+    perf_ptr_t perf1 = create();
+    add_func(perf1, 500, 2999, UCP_PROTO_PERF_FACTOR_LOCAL_TL, local_tl_func);
+    UCS_TEST_MESSAGE << perf1.get();
+
+    perf_ptr_t perf2 = create();
+    add_func(perf2, 1000, 1999, UCP_PROTO_PERF_FACTOR_LOCAL_CPU,
+             local_cpu_func);
+    UCS_TEST_MESSAGE << perf2.get();
+
+    m_perf = aggregate({perf1, perf2});
+    UCS_TEST_MESSAGE << m_perf.get();
+
+    expect_empty_range(0, 999);
+    expect_perf(1000, 1999,
+                {{UCP_PROTO_PERF_FACTOR_LOCAL_CPU, local_cpu_func},
+                 {UCP_PROTO_PERF_FACTOR_LOCAL_TL, local_tl_func}});
+    expect_empty_range(2000, SIZE_MAX);
+}
+
+UCS_TEST_P(test_proto_perf, agg3)
```

Comment:
do we have a case with multiple intersections being output with ranges overlaps? maybe add other ranges to generate multiple intersections.